chain/chain/src/garbage_collection.rs

```diff
@@ -1161,10 +1161,22 @@ fn gc_parent_shard_after_resharding(
     let shard_layout = epoch_manager.get_shard_layout(&next_epoch_id)?;
     let mut trie_store_update = store.trie_store().store_update();
     for parent_shard_uid in shard_layout.get_split_parent_shard_uids() {
-        // Delete the state of the parent shard
-        tracing::debug!(target: "garbage_collection", ?parent_shard_uid, "resharding state cleanup");
-        trie_store_update.delete_shard_uid_prefixed_state(parent_shard_uid);
+        // Check if any child shard still map to this parent shard
+        let has_active_mapping = shard_layout.shard_uids().any(|child_shard_uid| {
```

Comment:
nit: Use function `get_children_shards_uids` here.
```
        let has_active_mapping = shard_layout
            .get_children_shards_uids(parent_shard_uid.shard_id())
            .unwrap()
            .into_iter()
            .any(|child_shard_uid| {
                get_shard_uid_mapping(&store, child_shard_uid) == parent_shard_uid
            });
```

Reply:
If we do that we should be careful when "removing support for older versions". What would be the proper fix so eventually there can just be 1 code path? A database migration for nodes that did an old resharding?